controllers/skydiveagents_controller.go

```diff
@@ -0,0 +1,89 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	"k8s.io/client-go/tools/clientcmd"
+	"skydive/pkg/kclient"
+	"skydive/pkg/manifests"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	skydivev1beta1 "skydive/api/v1beta1"
+)
+
+var (
+	SkydiveAgentsDaemonSet = "skydive-agents/daemon-set.yaml"
+)
+
+// SkydiveAgentsReconciler reconciles a SkydiveAgents object
+type SkydiveAgentsReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=skydive.example.com,resources=skydiveagents,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=skydive.example.com,resources=skydiveagents/status,verbs=get;update;patch
+func (r *SkydiveAgentsReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("skydiveanalyzer", req.NamespacedName)
+
+	log.Info("initializing assets for skydive-agents")
+	assets := manifests.NewAssets("assets")
+
+	log.Info("Building configuration")
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
```

Comment:
Replace with

```	var config *rest.Config
	var err error
	if kubeConfig := os.Getenv("KUBECONFIG"); kubeConfig != "" {
		config, err = clientcmd.BuildConfigFromFlags("", kubeConfig)
	} else {
		config, err = rest.InClusterConfig()
	}```

Reply:
So in my case, I'll have to add an ENV var named KUBECONFIG and set its path right?

Reply:
@oranichu yes or if you are logged into a cluster using kubectl in the current shell you should also be covered